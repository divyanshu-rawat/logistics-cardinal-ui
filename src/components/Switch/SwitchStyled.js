import styled from 'styled-components';

const SwitchStyled = styled('div')`
  display: flex;
  align-items: center;
  label {
    margin-bottom: 0;
    margin-right: ${({ theme }) => theme.spacings.kilo};
  }
  .react-switch-bg {
    ${({ theme, active }) => {
      if (!active) {
        return `border: 1px solid ${theme.colors.grey200}`;
      }
      return '';
    }};
  }
`;

export default SwitchStyled;
