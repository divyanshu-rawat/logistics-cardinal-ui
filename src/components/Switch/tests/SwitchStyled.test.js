import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SwitchStyled from '../SwitchStyled';
import Theme from '../../../themes/rooster';

const renderComponent = (active = false) => (
  <SwitchStyled theme={Theme} active={active} />
);

describe('<SwitchStyled />', () => {
  it('should render the component', () => {
    const tree = toJson(shallow(renderComponent()));
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with `active state`', () => {
    const tree = toJson(shallow(renderComponent(true)));
    expect(tree).toMatchSnapshot();
  });
});
