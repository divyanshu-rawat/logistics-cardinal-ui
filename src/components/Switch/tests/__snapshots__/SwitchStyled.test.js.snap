// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwitchStyled /> should render the component 1`] = `
<div
  className="sc-bdVaJa jatKZq"
/>
`;

exports[`<SwitchStyled /> should render the component with \`active state\` 1`] = `
<div
  className="sc-bdVaJa iEKfuB"
/>
`;
